server/consumer.go

```diff
@@ -2327,17 +2344,22 @@ func (o *consumer) processAck(subject, reply string, hdr int, rmsg []byte) {
 	}
 
 	sseq, dseq, dc := ackReplyInfo(subject)
+	var err *ApiError
 
 	skipAckReply := sseq == 0
 
 	switch {
 	case len(msg) == 0, bytes.Equal(msg, AckAck), bytes.Equal(msg, AckOK):
-		if !o.processAckMsg(sseq, dseq, dc, reply, true) {
-			// We handle replies for acks in updateAcks
-			skipAckReply = true
+		if b, e := o.processAckMsg(sseq, dseq, dc, reply, true); e == nil {
+			if !b {
```

Comment:
This could optionally be simplified to `e == nil && !b`

Reply:
Done